corehq/apps/repeaters/models.py

```diff
@@ -301,7 +283,15 @@ def get_auth(self):
     def fire_for_record(self, repeat_record):
         headers = self.get_headers(repeat_record)
         auth = self.get_auth()
-        payload = self.get_payload_and_handle_exception(repeat_record)
+
+        try:
+            payload = self.get_payload(repeat_record)
+        except Exception as e:
+            # todo: seems like this just does everything that handle_exception does but not as well.
+            # todo:   seems like they could be combined
```

Comment:
You recently changed the behaviour of that method: https://github.com/dimagi/commcare-hq/pull/16059/commits/96539922436a57d1ec137cb93480b0260e41bde6#diff-7a94e70a3056d09745068e953a4847f6L558
It used to reraise the exception based on a flag, which the other handle_exception method didn't do. 

the `handle_payload_exception` method also didn't bubble up to call any `repeater` (or `repeater_generator`) handling methods. 

You are correct about the intention of this code to catch errors in the `get_payload` calls.